app/src/main/java/jaelson/wagner/silva/upcomingmovies/CrashReportingTree.java

```diff
@@ -0,0 +1,31 @@
+package jaelson.wagner.silva.upcomingmovies;
+
+/**
+ * Created by jaelsonwagner@gmail.com.
+ */
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+import timber.log.Timber;
+
+/**
+ * A tree which logs important information for crash reporting.
+ */
+public class CrashReportingTree extends Timber.Tree {
+  @Override
+  protected void log(int priority, String tag, @NonNull String message, Throwable t) {
+    if (priority == Log.VERBOSE || priority == Log.DEBUG) {
+      return;
+    }
+
+    SomeCrashLibrary.log(priority, tag, message);
```

Comment:
Interesting idea. Have you used crash reporting tools? Is this how you integated them?

Reply:
Yes. I have used Crashlytics. Making CrashReportingTree a wrapper logger class, allow us use our preferred crash reporting lib.